### ISSUE TITLE
Mismatch in filename and `export` in Talking to a backend:Updating song ratings

### ISSUE THREAD

**Issue by jluckyiv** [balinterdi/rarwe-issues#90]
The last code sample seems to be for the controller, but it exports a route:

```javascript
// app/controllers/bands/band/songs.js
export default Ember.Route.extend({
  (...)
  actions: {
    (...)
    updateRating: function(params) {
      // code
    } 
});
```

I think it should be `Ember.Controller.extend`.

**Comment by balinterdi** [comment]
Thank you, Jack, it should be `Ember.Controller`.

**Comment by starspace1** [comment]
If it's recommended to put code changing model state in routes, which is updateRating only handled in the details controller rather than bubbling up to the details route? Also, why does .save in updateRating send a PUT request while the save in the details route sends a PATCH request? Could be useful to touch on both topics in this section of the book.

**Comment by balinterdi** [comment]
1. Great question. The reason `updateRating` now lives in the controller is because I use a closure action in the template (`{{star-rating ... on-click=(action "updateRating")}}`) and closure actions are looked up/invoked on the context of the template, so I had to move it from the route to the controller. Putting actions that change global state on routes is more like a rule of thumb and the ability to use closure actions actually supersedes that rule, in my opinion. I should probably make that explicit.
2. `model.save()` went at one point from issuing a PUT to issuing a PATCH. I've changed that in the book now.

Thank you, Leslie!
